tagmanager/src/tm_parser.c

```diff
@@ -0,0 +1,647 @@
+};
+
+/* not in universal-ctags */
+static TMParserMapEntry nsis_map[] = {
+	{'n', tm_tag_namespace_t},
+	{'f', tm_tag_function_t},
+	{'v', tm_tag_variable_t},
+};
+
+/* not in universal-ctags */
+static TMParserMapEntry markdown_map[] = {
+	{'v', tm_tag_variable_t},
+};
+
+/* not in universal-ctags */
+static TMParserMapEntry txt2tags_map[] = {
+	{'m', tm_tag_member_t},
+};
+
+/* not in universal-ctags */
+static TMParserMapEntry abc_map[] = {
+	{'m', tm_tag_member_t},
+	{'s', tm_tag_struct_t},
+};
+
+static TMParserMapEntry verilog_map[] = {
+	{'c', tm_tag_variable_t},
+	{'e', tm_tag_typedef_t},
+	{'f', tm_tag_function_t},
+	{'m', tm_tag_class_t},
+	{'n', tm_tag_variable_t},
+	{'p', tm_tag_variable_t},
+	{'r', tm_tag_variable_t},
+	{'t', tm_tag_function_t},
+};
+
+static TMParserMapEntry r_map[] = {
+	{'f', tm_tag_function_t},
+	{'l', tm_tag_other_t},
+	{'s', tm_tag_other_t},
+};
+
+static TMParserMapEntry cobol_map[] = {
+	{'d', tm_tag_variable_t},
+	{'f', tm_tag_function_t},
+	{'g', tm_tag_struct_t},
+	{'p', tm_tag_macro_t},
+	{'P', tm_tag_class_t},
+	{'s', tm_tag_namespace_t},
+};
+
+static TMParserMapEntry objc_map[] = {
+	{'i', tm_tag_interface_t},
+	{'I', tm_tag_undef_t},
+	{'P', tm_tag_undef_t},
+	{'m', tm_tag_method_t},
+	{'c', tm_tag_class_t},
+	{'v', tm_tag_variable_t},
+	{'F', tm_tag_field_t},
+	{'f', tm_tag_function_t},
+	{'p', tm_tag_undef_t},
+	{'t', tm_tag_typedef_t},
+	{'s', tm_tag_struct_t},
+	{'e', tm_tag_enum_t},
+	{'M', tm_tag_macro_t},
+};
+
+/* not in universal-ctags */
+static TMParserMapEntry asciidoc_map[] = {
+	{'n', tm_tag_namespace_t},
+	{'m', tm_tag_member_t},
+	{'d', tm_tag_macro_t},
+	{'v', tm_tag_variable_t},
+	{'s', tm_tag_struct_t},
+};
+
+/* not in universal-ctags */
+static TMParserMapEntry abaqus_map[] = {
+	{'c', tm_tag_class_t},
+	{'m', tm_tag_member_t},
+	{'n', tm_tag_interface_t},
+};
+
+static TMParserMapEntry rust_map[] = {
+	{'n', tm_tag_namespace_t},
+	{'s', tm_tag_struct_t},
+	{'i', tm_tag_interface_t},
+	{'c', tm_tag_class_t},
+	{'f', tm_tag_function_t},
+	{'g', tm_tag_enum_t},
+	{'t', tm_tag_typedef_t},
+	{'v', tm_tag_variable_t},
+	{'M', tm_tag_macro_t},
+	{'m', tm_tag_field_t},
+	{'e', tm_tag_enumerator_t},
+	{'F', tm_tag_method_t},
+};
+
+static TMParserMapEntry go_map[] = {
+	{'p', tm_tag_namespace_t},
+	{'f', tm_tag_function_t},
+	{'c', tm_tag_macro_t},
+	{'t', tm_tag_typedef_t},
+	{'v', tm_tag_variable_t},
+	{'s', tm_tag_struct_t},
+	{'i', tm_tag_interface_t},
+	{'m', tm_tag_member_t},
+};
+
+static TMParserMapEntry json_map[] = {
+	{'o', tm_tag_member_t},
+	{'a', tm_tag_member_t},
+	{'n', tm_tag_member_t},
+	{'s', tm_tag_member_t},
+	{'b', tm_tag_member_t},
+	{'z', tm_tag_member_t},
+};
+
+/* not in universal-ctags */
+static TMParserMapEntry powershell_map[] = {
+	{'f', tm_tag_function_t},
+	{'v', tm_tag_variable_t},
+};
+
+
+typedef struct
+{
+    TMParserMapEntry *entries;
+    guint size;
+} TMParserMap;
+
+#define MAP_ENTRY(map) {map, (sizeof(map)/sizeof(TMParserMapEntry))}
```

Comment:
> Could do the macro stuff you suggest too, just to me it usually takes more time to decode what macros with ## do than not using such features ;-).

It sure hides some logic, but OTOH it can avoid typos in repetitive code, as well as making it easier to change it.
In this case, instead of doing `[TM_PARSER_C] = { map_C, G_N_ELEMENTS(map_C) }` we get `MAP_ENTRY(C)`, which, IMO, is worth it, esp. as it's very repetitive.

BTW, I see I made a typo in my previous post's macro, missing the `TM_PARSER_##` prefix to `lang`.

> After posting the patch I realized it would be probably nicer instead of doing […]

Indeed.  Can even be a simple `#define map_CPP map_C` avoiding any actual data duplication.